app.py

```diff
@@ -58,20 +58,29 @@ def main():
     if topic:
         model1_messages = []
         model2_messages = []
+        stop = False
         for _i in range(st.session_state.exchange):
```

Comment:
you're still using the `exchange` state here. I suggest you use a while loop on the stop condition:
```
while not stop:
     ....
```